src/modality-agnostic-files.md

```diff
@@ -11,7 +11,12 @@ Templates:
 
 ### `dataset_description.json`
 
-The file `dataset_description.json` is a JSON file describing the dataset.
+<!-- This block generates a description.
+A guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___render_description("files", "dataset_description") }}
```

Comment:
Do we want this to be so specific that it only works for `objects.{object}.{key}.description`? If so, I'd suggest just combining them:

```suggestion
{{ MACROS___render_object_description("files.dataset_description") }}
```

We could also make it more general, and allow any text field:

```suggestion
{{ MACROS___render_text("objects.files.dataset_description.description") }}
```

This could be implemented very easily:

```Python
def render_text(schema, key: str, src_path=None) -> str:
    text = schema.get(key)
    if not isinstance(text, str):
        raise ValueError(f"{key} does not refer to a text field")
    return text
```

Reply:
nice one

this also shows I don't know the schema code well enough

Reply:
The schema object is a namespace, and these are all the same:

* `schema.objects.files`
* `schema.objects['files']`
* `schema['objects.files']`
* `schema['objects']['files']`
* `schema.get('objects.files')`

There's also a `level` keyword for `keys`, `values` and `items`:

```python
>>> list(schema.keys(level=2))
['meta.context',
 'meta.expression_tests',
 'meta.versions',
 'objects.columns',
 'objects.common_principles',
 'objects.datatypes',
 'objects.entities',
 'objects.extensions',
 'objects.files',
 'objects.formats',
 'objects.metadata',
 'objects.modalities',
 'objects.suffixes',
 'rules.checks',
 'rules.common_principles',
 'rules.dataset_metadata',
 'rules.entities',
 'rules.errors',
 'rules.files',
 'rules.modalities',
 'rules.sidecars',
 'rules.tabular_data']
```

Reply:
done

Reply:
We still have a lot of `render_description()` calls to replace.

Reply:
indeed!
🙈 
this has been a long day

Reply:
You are allowed to sleep...

Reply:
well in the mean time the spec should be fixed now and render properly now